src/smartcontracts/typesystem/managedDecimal.ts

```diff
@@ -84,8 +89,8 @@ export class ManagedDecimalSignedValue extends TypedValue {
     private readonly value: BigNumber;
     private readonly scale: number;
 
-    constructor(value: BigNumber.Value, scale: number) {
-        super(new ManagedDecimalType(scale));
+    constructor(value: BigNumber.Value, scale: number, isVariableDecimals: boolean = false) {
+        super(new ManagedDecimalSignedType(isVariableDecimals ? "usize" : scale));
```

Comment:
:see_no_evil: 

Also in `ManagedDecimalValue`.

Is it worth to encapsulate this tricky logic somewhere? E.g. named constructor of the types?